compute_sdk/globus_compute_sdk/sdk/diagnostic.py

```diff
@@ -0,0 +1,497 @@
+                    conn_data = ssock.cipher()
+                    cipher, version, _ = conn_data if conn_data else (None, None, None)
+            print(
+                f"Successfully established SSL connection with {host}:{port}!\n"
+                f"Version: {version}\n"
+                f"Cipher:  {cipher}\n"
+            )
+        except OSError as e:
+            print_warning(
+                f"Failed to establish SSL connection with {host}:{port} - {e}\n",
+            )
+
+    kernel.display_name = f"test_ssl_conn({host}, {port})"  # type: ignore
+    return kernel
+
+
+def print_service_versions(base_url: str):
+    def kernel():
+        # Just adds a newline to the version info for better formatting
+        version_info = WebClient(base_url=base_url).get_version(service="all")
+        print(f"{version_info}\n")
+
+    kernel.display_name = f"get_service_versions({base_url})"  # type: ignore
+    return kernel
+
+
+def ifconfig_wrapper():
+    if sys.platform in ["win32"]:
+        return "ipconfig"
+
+    return "ifconfig"
+
+
+@display_name("OpenSSL version")
+def print_openssl_version():
+    print(f"{ssl.OPENSSL_VERSION}\n")
+
+
+def print_warning(s: str):
+    """
+    Prints both to stderr (goes to console) and to stdout (captured into file)
+    This method and print_highlight() below uses rich.Console.print() instead
+    of the base rich.print() as no_wrap defaults to None for stdout, inserting
+    line breaks when we don't want them (in tests for example)
+    """
+    Console(file=sys.stderr).print(f"[bold red]{s}[/bold red]", no_wrap=True)
+    print(s)
+
+
+def print_highlight(s: str):
+    get_console().print(f"[bold yellow]{s}[/bold yellow]", no_wrap=True)
+
+
+def os_info_cmd():
+    if sys.platform == "linux":
+        return cat("/etc/os-release")
+    elif sys.platform == "darwin":
+        return "sw_vers"
+    elif sys.platform == "win32":
+        return "systeminfo"
+    else:
+        return None
+
+
+def if_supported_on_os(cmd: str):
+    """
+    Misc commands that may exist for one or more platforms, but not for others.
+
+    A simpler 'supported or not' check that returns None if unavailable
+    """
+    if cmd:
+        BY_PLATFORM = {
+            "ip": ["linux"],
+            "lshw": ["linux"],
+            "df": ["linux", "darwin"],
+            "sysctl": ["linux", "darwin"],
+        }
+
+        cmd_base = cmd.split()[0]
+        if cmd_base in BY_PLATFORM:
+            return cmd if sys.platform in BY_PLATFORM.get(cmd_base, []) else None
+        else:
+            # If not specifically named here, assume it's runnable on all
+            return cmd
+    else:
+        return None
+
+
+def filter_by_os(cmd_list: list):
+    results = []
+    for cmd in cmd_list:
+        # if_supported_on_os only checks cmds in str form, not methods
+        if cmd and (not isinstance(cmd, str) or if_supported_on_os(cmd)):
+            results.append(cmd)
+
+    return results
```

Comment:
Yes that is cleaner and more centralized.  The reason I didn't do something similar is that I didn't want to change the existing hardware_report code much, so filtering it after getting the list instead of changing the original data structure required less refactoring.

I'll try this approach and see if it is easily done again.

Reply:
Not a sticking point; your call.